src/validation/__tests__/OverlappingFieldsCanBeMerged-test.js

```diff
@@ -938,4 +938,55 @@ describe('Validate: Overlapping fields can be merged', () => {
 
   });
 
+  it('does not infinite loop on recursive fragment', () => {
+    expectPassesRule(OverlappingFieldsCanBeMerged, `
+      fragment fragA on Human {name relatives { name ...fragA } },
+    `);
+  });
+
+  it('does not infinite loop on immediately spread fragment', () => {
+    expectPassesRule(OverlappingFieldsCanBeMerged, `
+      fragment fragA on Human {name ...fragA },
```

Comment:
add a test for immediate spreads but in a larger cycle

fragment fragA on Human {name ...fragB }
fragment fragB on Human {name ...fragA }